### DISCUSSION TITLE
Reading a slice of an arrays across PyTrees

### DISCUSSION THREAD

**Discussion by darsnack** [google/orbax#769] [Q&A]
I have a checkpoint which is a single PyTree. Each leaf has a leading batch dimension. I want to be able to restore the `i`-th slice of this leading dimension only. The end result should be a PyTree of the same structure as the checkpoint, but each leaf is the `i`-th slice. Is there a way to write a custom handler to facilitate this? I know that Zarr3 let's you read only a chunk of the data from disk.

**Comment by niketkumar** [comment]
Please consider the following options:

* [PyTreeRestoreArgs.transforms](https://github.com/google/orbax/blob/d5cf598c3f400ea5c5d1fc044bd10535488cbc67/checkpoint/orbax/checkpoint/pytree_checkpoint_handler.py#L1519) in [PyTreeCheckpointHandler.restore](https://orbax.readthedocs.io/en/latest/api_reference/checkpoint.checkpoint_handlers.html#orbax.checkpoint.PyTreeCheckpointHandler.restore).

* Write a custom `TypeHandler` to handle `ndarray` or `jax.Array`. See https://orbax.readthedocs.io/en/latest/custom_handlers.html#typehandler for details.
